markdown/extensions/abbr.py

```diff
@@ -69,13 +93,14 @@ def iter_element(self, el: etree.Element, parent: etree.Element | None = None) -
             self.iter_element(child, el)
         if text := el.text:
             for m in reversed(list(self.RE.finditer(text))):
-                abbr = etree.Element('abbr', {'title': self.abbrs[m.group(0)]})
-                abbr.text = AtomicString(m.group(0))
-                abbr.tail = text[m.end():]
-                el.insert(0, abbr)
-                text = text[:m.start()]
+                if self.abbrs[m.group(0)]:
```

Comment:
Curious why you felt the need to wrap this in an if statement. Seem reasonable to expect that if we have a match this should always be True. If there is a specific case where it is not, then we need a test case for it.

Reply:
This handles the situation in `test_abbr_empty()` of declaring an abbreviation without a definition. Without that test it adds `abbr` tags with an empty title instead of leaving it unmodified.

Reply:
Ah, right. I wasn't paying attention to what you were actually testing. You are checking the value, not the key. And I suppose a user could pass in a glossary with keys which contain empty values. Yes, this needs to stay here.